packages/cli-utils/src/index.ts

```diff
@@ -1,58 +1,101 @@
 import sade from 'sade';
 import { promises as fs, existsSync } from 'node:fs';
-import path from 'node:path';
+import path, { resolve } from 'node:path';
 // We use comment-json to parse the tsconfig as the default ones
 // have comment annotations in JSON.
 import { parse } from 'comment-json';
+import type { IntrospectionQuery } from 'graphql';
+import { buildClientSchema, getIntrospectionQuery, printSchema } from 'graphql';
 import type { TsConfigJson } from 'type-fest';
+
+import type { GraphQLSPConfig } from './lsp';
+import { hasGraphQLSP } from './lsp';
 import { ensureTadaIntrospection } from './tada';
 
 const prog = sade('gql.tada');
 
 prog.version(process.env.npm_package_version || '0.0.0');
 
-type GraphQLSPConfig = {
-  name: string;
-  schema: string;
-  tadaOutputLocation?: string;
-};
-
-function hasGraphQLSP(tsconfig: TsConfigJson): boolean {
-  if (!tsconfig.compilerOptions) {
-    console.warn('Missing compilerOptions object in tsconfig.json.');
-    return false;
-  }
+async function main() {
+  prog
+    .command('generate-schema <target> <dest>')
+    .describe(
+      'Generate a GraphQL schema from a URL or introspection file, this will be generated from the parameters to this command.'
+    )
+    .option('--header', 'Pass a header to be used when fetching the introspection.')
+    .example("generate-schema https://example.com --header 'Authorization: Bearer token'")
+    .example('generate-schema ./introspection.json')
+    .action(async (target, destination, options) => {
```

Comment:
It probably makes sense to read the destination from the `tsconfig.json` (similar to: https://github.com/0no-co/gql.tada/blob/b09e2649ec9e6fd57b4290365ec1e1951058e12c/packages/cli-utils/src/index.ts#L86)

Both commands could later on allow for an override option (e.g. `-o --output`, I guess. But we have the benefit of knowing where that schema should end up